qurro/tests/test_qarcoal.py

```diff
@@ -155,24 +220,223 @@ def test_samples_to_use(self, get_mp_data):
         assert q.shape[0] == num_gut_samples
 
 
-def test_large_numbers():
-    # Qurro fails when x > |2^53 - 1| due to JS implementation
-    bignum = 2 ** 53 - 1
-    large_vals = [np.random.randint(bignum + 1, bignum * 2) for x in range(30)]
-    mat = np.reshape(large_vals, (6, 5))
-    samps = ["S{}".format(i) for i in range(5)]
-    feats = ["F{}".format(i) for i in range(6)]
-    table = biom.table.Table(mat, feats, samps)
-    tax_labels = [
-        "Bulbasaur",
-        "Ivysaur",
-        "Venusaur",
-        "Charmander",
-        "Charmeleon",
-        "Charizard",
-    ]
-    confidence = ["0.99"] * 6
-    taxonomy = pd.DataFrame([feats, tax_labels, confidence]).T
-    taxonomy.columns = ["feature-id", "Taxon", "Confidence"]
-    taxonomy.set_index("feature-id", inplace=True, drop=True)
-    qarcoal(table, Metadata(taxonomy), "Char", "saur")
+class TestIrregularData():
+    @pytest.fixture(scope="class")
+    def get_testing_data(self):
+        mat = [np.random.randint(1, 10) for x in range(30)]
+        mat = np.reshape(mat, (6, 5))
+        samps = ["S{}".format(i) for i in range(5)]
+        feats = ["F{}".format(i) for i in range(6)]
+        table = biom.table.Table(mat, feats, samps)
+        tax_labels = [
+            "Bulbasaur",
+            "Ivysaur",
+            "Venusaur",
+            "Charmander",
+            "Charmeleon",
+            "Charizard",
+        ]
+        confidence = ["0.99"] * 6
+        taxonomy = pd.DataFrame([
+            feats,
+            tax_labels,
+            confidence,
+        ]).T
+        taxonomy.columns = ["feature-id", "Taxon", "Confidence"]
+        taxonomy.set_index("feature-id", inplace=True, drop=True)
+        data = namedtuple("Data", "table taxonomy")
+        return data(table, taxonomy)
+
+    def test_taxonomy_missing_features(self, get_testing_data):
+        """Taxonomy file missing features that are present in feature table"""
+        table = get_testing_data.table.to_dataframe()
```

Comment:
Oh wow, you're right. I think I initially was planning on writing some tests that call `qarcoal` directly where the BIOM would be more input directly. Since that didn't end up being the case, this is an easy fix to make.